posthog-core/src/index.ts

```diff
@@ -0,0 +1,554 @@
+      },
+    })
+
+    this.enqueue('autocapture', payload)
+    return this
+  }
+
+  /***
+   *** GROUPS
+   ***/
+
+  groups(groups: { [type: string]: string }): this {
+    // Get persisted groups
+    const existingGroups = this.props.$groups || {}
+
+    // NOTE: Should we do the same for groups listed in identify / capture?
+    this.register({
+      $groups: {
+        ...existingGroups,
+        ...groups,
+      },
+    })
+
+    if (Object.keys(groups).find((type) => existingGroups[type] !== groups[type]) && this._decideResponsePromise) {
+      void this.reloadFeatureFlagsAsync()
+    }
+
+    return this
+  }
+
+  group(groupType: string, groupKey: string, groupProperties?: PostHogEventProperties): this {
+    this.groups({
+      [groupType]: groupKey,
+    })
+
+    if (groupProperties) {
+      this.groupIdentify(groupType, groupKey, groupProperties)
+    }
+
+    return this
+  }
+
+  groupIdentify(groupType: string, groupKey: string, groupProperties?: PostHogEventProperties): this {
+    const payload = {
+      event: '$groupidentify',
+      distinctId: `$${groupType}_${groupKey}`,
+      properties: {
+        $group_type: groupType,
+        $group_key: groupKey,
+        $group_set: groupProperties || {},
+        ...this.getCommonEventProperties(),
+      },
+    }
+
+    this.enqueue('capture', payload)
+    return this
+  }
+
+  /***
+   *** FEATURE FLAGS
+   ***/
+  private decideAsync(): Promise<PostHogDecideResponse> {
+    if (this._decideResponsePromise) {
+      return this._decideResponsePromise
+    }
+    return this._decideAsync()
+  }
+
+  private async _decideAsync(): Promise<PostHogDecideResponse> {
+    const url = `${this.host}/decide/?v=2`
+
+    const distinctId = this.getDistinctId()
+    const groups = this.props.$groups || {}
+
+    const fetchOptions: PostHogFetchOptions = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({ groups, distinct_id: distinctId, token: this.apiKey }),
+    }
+
+    this._decideResponsePromise = this.fetchWithRetry(url, fetchOptions)
+      .then((r) => r.json() as PostHogDecideResponse)
+      .then((res) => {
+        if (res.featureFlags) {
+          this.setPersistedProperty<PostHogDecideResponse['featureFlags']>(
+            PostHogPersistedProperty.FeatureFlags,
+            res.featureFlags
+          )
+        }
+
+        this._events.emit('featureflags', res.featureFlags)
+        return res
+      })
+    return this._decideResponsePromise
+  }
+
+  getFeatureFlag(key: string, defaultResult: string | boolean = false): boolean | string | undefined {
+    const featureFlags = this.getFeatureFlags()
+
+    if (!featureFlags) {
+      // If we haven't loaded flags yet we respond undefined to indicate this
+      return undefined
+    }
+
+    if (this.sendFeatureFlagEvent && !this.flagCallReported[key]) {
+      this.flagCallReported[key] = true
+      this.capture('$feature_flag_called', {
```

Comment:
This is how it also works in posthog-js but @neilkakkar how come we're sending this capture call in the server libraries regardless of whether we're sending `sendFeatureFlag` but in the client side we only do this under the condition?

Reply:
And we don't have to send the feature flag properties in the `capture` call here

Reply:
> And we don't have to send the feature flag properties in the `capture` call here

I don't quite follow what you mean by this?

Reply:
>  @neilkakkar how come we're sending this capture call in the server libraries regardless of whether we're sending sendFeatureFlag but in the client side we only do this under the condition?

good catch, I wasn't aware that posthog-js doesn't send it with every call, but once per user per library instantiation.

Reply:
Since it doesn't affect experiments whether it's once or more than once per user, I guess makes sense for server-side to follow the same.

Reply:
@benjackwhite I think you're good here! Just some confusion I had with server vs client side implementation of the feature flag called 😅